nedlibreader/time.go

```diff
@@ -0,0 +1,31 @@
+package nedlibreader
+
+import (
+	"fmt"
+	"time"
+)
+
+func parseTime(dateString string) (parsedTime time.Time, err error) {
+	parsedTime, err = time.Parse(time.RFC1123, dateString)
+	if err == nil {
+		return
+	}
+
+	parsedTime, err = time.Parse(time.RFC850, dateString)
+	if err == nil {
+		return
+	}
+
+	parsedTime, err = time.Parse(time.ANSIC, dateString)
+	if err == nil {
+		return
+	}
+
+	parsedTime, err = time.Parse(time.UnixDate, dateString)
+	if err == nil {
+		return
+	}
+
+	err = fmt.Errorf("failed to parse string as time.Time: %s: %v", dateString, err)
```

Comment:
```suggestion
	err = fmt.Errorf("failed to parse string as time.Time: %s: %w", dateString, err)
```

Wrap error to be able to use `errors.Is` or `errors.As`.

Reply:
This function will return the error message from the latest attempt which is a bit misleading considering we are trying different layouts.

Maybe we should return

```
time.ParseError{
    Message: ": cannot parse as [list of formats?]" 
}
```

considering:

```
// Error returns the string representation of a ParseError.
func (e *ParseError) Error() string {
	if e.Message == "" {
		return "parsing time " +
			quote(e.Value) + " as " +
			quote(e.Layout) + ": cannot parse " +
			quote(e.ValueElem) + " as " +
			quote(e.LayoutElem)
	}
	return "parsing time " +
		quote(e.Value) + e.Message
}
```

ref:  https://cs.opensource.google/go/go/+/refs/tags/go1.22.2:src/time/format.go;drc=3ad6393f8676b1b408673bf40b8a876f29561eef;l=885

Reply:
The error message is misleading, I agree. We should probably improve it, but right now the current implementation of this PR is better than what exists in `main`.

I added a fixup to fix the error handling in the invalid time format cases, so now we only check for the type, not the actual error message.